bin/p2-rm/main.go

```diff
@@ -24,183 +20,90 @@ import (
 // Command line flags
 var (
 	podName      = kingpin.Arg("pod", "The names of the pod to be removed").Required().String()
-	nodeName     = kingpin.Flag("node", "The node to do the scheduling on. Uses the hostname by default.").String()
+	nodeName     = kingpin.Flag("node", "The node to unschedule the pod from. Uses the hostname by default. Only applies to \"legacy\" pods.").String()
+	podUniqueKey = kingpin.Flag("pod-unique-key", "The pod unique key to unschedule. Only applies to \"uuid\" pods. Cannot be used with --node").Short('k').String()
 	deallocation = kingpin.Flag("deallocate", "Specifies that we are deallocating this pod on this node. Using this switch will mutate the desired_replicas value on a managing RC, if one exists.").Bool()
 )
 
-type store interface {
-	NewSession(name string, renewalCh <-chan time.Time) (kp.Session, chan error, error)
-	DeletePod(podPrefix kp.PodPrefix, nodename types.NodeName, podId types.PodID) (time.Duration, error)
-}
-
-// P2RM contains the state necessary to safely remove a pod from a node
-type P2RM struct {
-	Store   store
-	RCStore rcstore.Store
-	Client  consulutil.ConsulClient
-	Labeler labels.ApplicatorWithoutWatches
-
-	LabelID  string
-	NodeName types.NodeName
-	PodName  string
-}
-
-// NewP2RM is a constructor for the P2RM type. It will generate the necessary
-// storage types based on its api.Client argument
-func NewP2RM(client consulutil.ConsulClient, podName string, nodeName types.NodeName, labeler labels.ApplicatorWithoutWatches) *P2RM {
-	rm := &P2RM{}
-	rm.Client = client
-	rm.Store = kp.NewConsulStore(client)
-	rm.RCStore = rcstore.NewConsul(client, labeler, 5)
-	rm.Labeler = labeler
-	rm.LabelID = path.Join(nodeName.String(), podName)
-	rm.PodName = podName
-	rm.NodeName = nodeName
-
-	return rm
-}
-
 func main() {
 	kingpin.Version(version.VERSION)
 	_, opts, labeler := flags.ParseWithConsulOptions()
 
+	consulClient := kp.NewConsulClient(opts)
+	if *podUniqueKey != "" {
+		err := handleUUIDPodRemoval(consulClient, types.PodUniqueKey(*podUniqueKey), types.PodID(*podName), labeler)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, err.Error())
+			os.Exit(1)
+		}
+	}
+
+	err := handleLegacyPodRemoval(consulClient, labeler)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, err.Error())
+		os.Exit(1)
+	}
+}
+
+func handleLegacyPodRemoval(consulClient consulutil.ConsulClient, labeler labels.ApplicatorWithoutWatches) error {
 	if *nodeName == "" {
 		hostname, err := os.Hostname()
```

Comment:
Is this behaviour all that helpful? I didn't find `p2-rm` installed in the datacenter so I can't imagine a use case for it.